contracts/p1/StRSR.sol

```diff
@@ -261,46 +262,41 @@ abstract contract StRSRP1 is Initializable, ComponentP1, IStRSR, EIP712Upgradeab
         require(rsrAmount <= rsrBalance, "Cannot seize more RSR than we hold");
         if (rsrBalance == 0) return;
 
-        // Calculate dust RSR threshold, the point at which we might as well call it a wipeout
-        // TODO: wait, that really, really doesn't pass dimensional analysis. wtf?
-        // dustRSRAmt:{qRSR} = D18 * {qDrafts + qStRSR} / D18
-        uint256 dustRSRAmt = (MIN_EXCHANGE_RATE * (totalDrafts + totalStakes)) / FIX_ONE; // {qRSR}
         uint256 seizedRSR;
-        if (rsrBalance <= rsrAmount + dustRSRAmt) {
-            // Rebase event: total RSR stake wipeout
-            seizedRSR = rsrBalance;
+        uint256 rewards = rsrRewards();
+
+        // Remove RSR from stakeRSR
+        uint256 stakeRSRToTake = (stakeRSR * rsrAmount + (rsrBalance - 1)) / rsrBalance;
+        stakeRSR -= stakeRSRToTake;
+        seizedRSR = stakeRSRToTake;
+
+        // update stakeRate, possibly beginning a new stake era
+        if (stakeRSR > 0) {
+            stakeRate = uint192((FIX_ONE_256 * totalStakes) / stakeRSR);
```

Comment:
Isn't it possible for `stakeRSR` to be arbitrarily small, such that this downcast truncates?